src/legacy/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory_provider.ts

```diff
@@ -28,7 +29,11 @@ export function visualizeEmbeddableFactoryProvider(Private: IPrivate) {
     savedVisualizations: SavedVisualizations,
     config: Legacy.KibanaConfig
   ) => {
-    return new VisualizeEmbeddableFactory(savedVisualizations, config);
+    return new VisualizeEmbeddableFactory(
+      savedVisualizations,
+      config,
+      Private(VisTypesRegistryProvider as any)
```

Comment:
~Shouldn't we be able to define it at least as `() => VisTypeRegistry`?~

Haven't you a bit down actually converted that file to TS, so we could plainly import the `VisTypesRegistryProvider` type here?